auslib/db.py

```diff
@@ -978,10 +979,17 @@ def select(self, where=None, transaction=None, **kwargs):
     def insert(self, changed_by, transaction=None, dryrun=False, **columns):
         base_columns, condition_columns = self._splitColumns(columns)
 
+        if base_columns["change_type"] == "delete":
```

Comment:
Unfortunately, at this level you can't assume that "change_type" will be present in columns. It certainly should be in all existing uses of it, but we don't know what will happen in the future. Your best bet is to explicitly check for it before this, and raise a ValueError if it's not present. You can see something similar at https://github.com/mozilla/balrog/blob/37b499a7d0c3d117186e3341eec7e2ea8f6e972d/auslib/db.py#L1501.